### PR TITLE
[GD-79] Fix event filter request

### PR DESCRIPTION
**Pull request by sidorovkirill**

### CODE REVIEW

File: Assets/AnkrSDK/Scripts/Core/Data/EventFilterRequest.cs [hunk 1]
```diff
@@ -1,11 +1,76 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Nethereum.ABI.FunctionEncoding.Attributes;
 using Nethereum.RPC.Eth.DTOs;
 
 namespace AnkrSDK.Core.Data
 {
-	public class EventFilterRequest<TEvDTO>
+	public class EventFilterRequest<TEvDto>
```
Comment (Vukiz) [inline Assets/AnkrSDK/Scripts/Core/Data/EventFilterRequest.cs:10]: DTO. - add it to you personal exception

File: Assets/AnkrSDK/Scripts/Core/Data/EventFilterRequest.cs [hunk 2]
```diff
@@ -1,11 +1,76 @@
...
-		public BlockParameter toBlock { get; set; }
+		private class Topic
+		{
+			public string Name { get; set; }
+			public string Alias { get; set; }
+			public object Value { get; set; }
+		}
+
+		public BlockParameter FromBlock { get; set; }
+		public BlockParameter ToBlock { get; set; }
```
Comment (Vukiz) [inline Assets/AnkrSDK/Scripts/Core/Data/EventFilterRequest.cs:20]: move public below private readonly

File: Assets/AnkrSDK/Scripts/Core/Data/EventFilterRequest.cs [hunk 3]
```diff
@@ -1,11 +1,76 @@
...
+		private readonly List<Topic> _topics;
```
Comment (Vukiz) [inline Assets/AnkrSDK/Scripts/Core/Data/EventFilterRequest.cs:21]: Can you make it IEnumerable<Topic>?  instead of List

**Review by Vukiz — CHANGES_REQUESTED** [review]

File: Assets/AnkrSDK/Scripts/Core/Data/EventFilterRequest.cs (hunk 2 above)
Comment (sidorovkirill) [reply Assets/AnkrSDK/Scripts/Core/Data/EventFilterRequest.cs:21]: Can you clarify why? Last time you said about that sequence if i'm right:
public
private
private readonly

Comment (Vukiz) [reply Assets/AnkrSDK/Scripts/Core/Data/EventFilterRequest.cs:22]: it is in reverse order
1 private readonly
2 private
3 public

File: Assets/AnkrSDK/Examples/Scripts/EventListenerExample/EventListenerExample.cs [hunk 4]
```diff
@@ -45,6 +45,21 @@ public async UniTaskVoid SubscribeWithTopics()
 				(TransferEventDTO t) => ReceiveEvent(t)
 			);
 		}
+		
+		// If you know only topic name then you can use EventFilterRequest
+		public async UniTaskVoid SubscribeWithRequest()
+		{
+			var filtersRequest = new EventFilterRequest<TransferEventDTO>();
+			filtersRequest.SetFromBlock(BlockParameter.CreateLatest());
```
Comment (Vukiz) [inline Assets/AnkrSDK/Examples/Scripts/EventListenerExample/EventListenerExample.cs:53]: Should we also use Earliest here?

File: Assets/AnkrSDK/Scripts/Core/Data/EventFilterRequest.cs [hunk 5]
```diff
@@ -1,11 +1,91 @@
...
+			{
+				var parameterAttribute = property.GetCustomAttribute<ParameterAttribute>(true);
+
+				return new Topic
+				{
+					Name = property.Name,
+					Alias = parameterAttribute.Name,
+					Values = new List<object>()
+				};
+			}).ToList();
```
Comment (Vukiz) [inline Assets/AnkrSDK/Scripts/Core/Data/EventFilterRequest.cs:88]: no need to return the list here. Just store IEnumerable?

**Review by Vukiz — CHANGES_REQUESTED** [review]

File: Assets/AnkrSDK/Examples/Scripts/EventListenerExample/EventListenerExample.cs (hunk 4 above)
Comment (sidorovkirill) [reply Assets/AnkrSDK/Examples/Scripts/EventListenerExample/EventListenerExample.cs]: Actualy it's no need here. FromBlocks works only for request past events

File: Assets/AnkrSDK/Scripts/Core/Data/EventFilterRequest.cs [hunk 6]
```diff
@@ -1,11 +1,91 @@
...
+		
+		public BlockParameter FromBlock { get; set; }
+		public BlockParameter ToBlock { get; set; }
+
+		public EventFilterRequest()
+		{
+			_topics = CollectTopics();
+		}
+
+		public void SetFromBlock(BlockParameter fromBlock)
```
Comment (Vukiz) [inline Assets/AnkrSDK/Scripts/Core/Data/EventFilterRequest.cs:29]: why do we need setter methods if we have public set for FromBlock/ToBlock properties?

**Review by Vukiz — CHANGES_REQUESTED** [review]

File: Assets/AnkrSDK/Scripts/Core/Data/EventFilterRequest.cs (hunk 6 above)
Comment (sidorovkirill) [reply Assets/AnkrSDK/Scripts/Core/Data/EventFilterRequest.cs:29]: I thought that one way of setting variables will be better. I mean if we have AddTopic why we need to recomend set another variables directly.

**Review by Vukiz — APPROVED** [review]

**Pull request merged**